plugins/autocertkit

```diff
@@ -401,6 +403,101 @@ def run_ack_logrotate(session, args):
 
     return json_dumps(msg)
 
+
+############### TCPDUMP Setup #########################
+
+def install_tcpdump(session, vm_ref, ip, username, password):
+    """Install tcpdump rpm on a droid vm"""
+    call = ['/usr/bin/scp',
+            "-o", "StrictHostKeyChecking=no",
+            "/opt/xensource/packages/files/auto-cert-kit/%s" % TCPDUMP_RPM,
+            'root@%s:' % ip]
+    log.debug(call)
+    make_local_call(call)
+    cmd_str = "rpm -i /root/%s" % TCPDUMP_RPM
+    log.debug(cmd_str)
+    ssh_command(ip, username, password, cmd_str)["stdout"]
+
+@log_exceptions
+def deploy_tcpdump(session, args):
+    """Install tcpdump on a specified VM"""
+    log.debug("deploy tcpdump plugin call")
+    vm_ref = validate_exists(args, 'vm_ref')
+
+    if not session.xenapi.VM.get_is_control_domain(vm_ref):
+        mip = validate_exists(args, 'mip')
+        username = validate_exists(args, 'username')
+        password = validate_exists(args, 'password')
+        # Only setup tcpdump if not control domain
+        # - that already has tcpdump installed.
+        i = 0
+        limit = 10
+        while i < limit:
+            try:
+                install_tcpdump(session, vm_ref, mip, username, password)
+                break
+            except:
+                log.debug(
+                    "Exception raised attempting to deploy tcpdump. Attempt %d" % i)
+                time.sleep(8)  # Sleep 8 seconds
+                i = i + 1  # Increment counter
+                if i == limit:
+                    raise PluginError("Plugin Error: Cannot deploy tcpdump")
+    else:
+        call = fr'''rpm -q tcpdump || rpm -i {INSTALL_DIR}/{TCPDUMP_RPM}'''
+        make_local_call(call, shell=True)
+
+    log.debug("tcpdump installation completed")
+    return json_dumps("OK")
+
+@log_exceptions
+def start_tcpdump_server(session, args):
+    """Start up tcpdump on the server. """
+
+    ip = validate_exists(args, 'ip')
+    iface = validate_exists(args, 'iface')
+
+    kill_processes('tcpdump')
+
+    # Check for tcpdump existence
+    if not process_running(TCPDUMP):
+        log.debug("tcpdump is not running - start process")
+
+        # Start tcpdump - need to ensure we don't block
+        call = "nohup /usr/sbin/tcpdump -s 128 -n -i %s src host %s -w tcpdump.pcap &" % (iface, ip)
+        log.debug("About to start tcpdump server (%s)" % call)
+        subprocess.Popen(call, shell=True)
```

Comment:
call `make_local_call()`

Reply:
make_local_call waits for the process return which would block the main process.